input/kafkamdm/kafkamdm.go

```diff
@@ -218,6 +218,27 @@ func (k *KafkaMdm) Start(handler input.Handler) {
 	}
 }
 
+// tryGetOffset will query kafka repeatedly for the requested offset, and return it when it has it,
+// or halt the program if can't retrieve the value
+func (k *KafkaMdm) tryGetOffset(topic string, partition int32, offsetStr string, offset int64, attempts int, sleep time.Duration) int64 {
+	var val int64
+	var err error
+	attempt := 1
+	for {
+		val, err = k.client.GetOffset(topic, partition, offset)
+		if err == nil {
+			break
+		}
+		if attempt == 7 {
+			log.Fatal(3, "kafka-mdm failed to get %s of partition %s:%d. %s (attempt %d/7)", offsetStr, topic, partition, err)
```

Comment:
I think you should have this method return an error instead and let the caller decide if the error is fatal

Reply:
>  if the user has MT confured to resume from "last" but the last offset stored by offsetMger is no longer available in kafka

isn't the behavior here par with what's in master? AFAICT, master will also crash because it has this:
```
        pc, err := k.consumer.ConsumePartition(topic, partition, currentOffset)
        if err != nil {
                log.Fatal(4, "kafka-mdm: failed to start partitionConsumer for %s:%d. %s", topic, partition, err)
        }
```

I think crashing in this case is the right thing to do.

Reply:
Internal libraries should (almost) never just crash, they should return an error indicating they can't continue and let the code that's calling them decide what to do about it.  What is the rationale for allowing the input processor to just stop the whole program dead in its tracks?

Reply:
the rationale for why to abort is discussed at https://github.com/raintank/metrictank/issues/520 ; this is also what i was referring to when i said the behavior is on par with master. (we've been doing this ever since kafka support was introduced)
as for why is the abort coming from the input plugin and not from `main`, i generally agree with you and am open to refactoring that. but let's do that in a future PR.  I want these fixes to get sooner rather than later.